v0/destinations/serenytics/transform.js

```diff
@@ -0,0 +1,220 @@
+  const STORAGE_URL = Config.storageUrlTrack;
+  const storageUrlEventMapping = getHashFromArrayWithDuplicate(
+    Config.eventToStorageUrlMap,
+    "from",
+    "to",
+    false
+  );
+  const { event } = message;
+  if (!event) {
+    throw new CustomError(
+      `[Serenytics]: event name is required in track call.`,
+      400
+    );
+  }
+  if (!storageUrlEventMapping[event] && !STORAGE_URL) {
+    throw new CustomError(`[Serenytics]: storage url is required.`, 400);
+  }
+  const productList = message.properties?.products;
+  if (productList && Array.isArray(productList)) {
+    const responseList = [];
+    const finalPayload = payload;
+    productList.forEach(product => {
+      const productDetails = constructPayload(
+        product,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.PROPERTY.name]
+      );
+      payload = { ...payload, ...productDetails };
+
+      const storageUrlEventList = storageUrlEventMapping[event];
+      if (storageUrlEventList) {
+        storageUrlEventList.forEach(eventUrl => {
+          const response = responseBuilder(eventUrl, payload);
+          responseList.push(response);
+        });
+      }
+      payload = finalPayload;
+    });
+    return responseList;
+  }
+
+  const storageUrlEventList = storageUrlEventMapping[event];
+  if (storageUrlEventList) {
+    const responseList = [];
+    storageUrlEventList.forEach(eventUrl => {
+      const response = responseBuilder(eventUrl, payload);
+      responseList.push(response);
+    });
+    return responseList;
+  }
+  const response = responseBuilder(STORAGE_URL, payload);
+  return response;
+};
+
+const processEvent = async (message, destination) => {
+  if (!message.type) {
+    throw new CustomError(
+      "Message Type is not present. Aborting message.",
+      400
+    );
+  }
+  const messageType = message.type.toLowerCase();
+  const { Config } = destination;
+  let STORAGE_URL;
+  let response;
+  let payload;
+  switch (messageType) {
+    case EventType.TRACK:
+      payload = payloadBuilder(
+        message,
+        CONFIG_CATEGORIES.TRACK.name,
+        ["properties"],
+        SERENYTICS_TRACK_EXCLUSION_LIST
+      );
+      response = await trackResponseBuilder(message, destination, payload);
+      return response;
+    case EventType.IDENTIFY:
+      if (!Config.storageUrlIdentify) {
+        throw new CustomError(`[Serenytics]: storage url is required.`, 400);
+      }
+      STORAGE_URL = Config.storageUrlIdentify;
+      payload = payloadBuilder(
+        message,
+        CONFIG_CATEGORIES.IDENTIFY.name,
+        ["traits", "context.traits"],
+        SERENYTICS_IDENTIFY_EXCLUSION_LIST
+      );
+      break;
+    case EventType.GROUP:
+      if (!Config.storageUrlGroup) {
+        throw new CustomError(`[Serenytics]: storage url is required.`, 400);
+      }
+      STORAGE_URL = Config.storageUrlGroup;
+      payload = payloadBuilder(
+        message,
+        CONFIG_CATEGORIES.GROUP.name,
+        ["properties"],
```

Comment:
For group call different attributes of group are called traits. Why are we providing `properties` here?

Reply:
In rudderstack docs, it is written that it is taken from `properties`
ref: https://www.rudderstack.com/docs/event-spec/standard-events/group/#:~:text=after%20removing%20the-,Common%20fields,-is%20as%20shown

Reply:
I took it as `track`, updated the code.